e3sm_to_cmip/util.py

```diff
@@ -306,195 +359,389 @@ def print_var_info(handlers, freq=None, inpath=None, tables=None, outpath=None):
                 if not has_vars:
                     continue
                 messages.append(msg)
-    
-    
+
     if outpath is not None:
-        with open(outpath, 'w') as outstream:
+        with open(outpath, "w") as outstream:
             yaml.dump(messages, outstream)
     else:
         pprint(messages)
+
+
 # ------------------------------------------------------------------
 
-def get_table(table, variable, freq, tables):
+
+def _load_handlers(
+    handlers_path: str,
+    tables_path: str,
+    var_list: List[str],
+    freq: str = "mon",
+    realm: str = "atm",
+    simple: bool = False,
+) -> List[Dict[str, Any]]:
+    """Loads default and/or complex handlers based on a list of variable names.
+
+    If this function is called with `simple=True`, then the `freq` argument
+    is not considered. This means that the base table is used (usually monthly).
+
+    Parameters
+    ----------
+    handlers_path : str
+        The path to the complex handlers, which are Python modules.
+    tables_path : str
+        The path to the CMIP6 tables.
+    var_list : List[str]
+        The list of variables to CMOR-ize.
+    freq : str, optional
+        The frequency, by default "mon".
+    realm : str, optional
+        The realm, by default "atm".
+    simple : bool, optional
+        If True, run in simple mode, by default False
+
+    Returns
+    -------
+    List[Dict[str, str]]:
+        A list of dictionaries, with each dictionary storing information related
+        to the CMOR handler of each variable.
     """
-    given the base table (for monthly data), find the correct
-    table given the frequency. Check the tables directory that this table
-    exists and the variable is included
+    available_handlers = _get_available_handlers(handlers_path)
+    chosen_handlers: List[Dict[str, Any]] = []
+
+    if "all" in var_list:
+        for key, handler in available_handlers.items():
+            # Update the handler's "table" if the frequency is not monthly.
+            if not simple and freq != "mon":
+                available_handlers[key]["table"] = _get_table_for_non_monthly_freq(
+                    handler["name"], handler["table"], freq, realm, tables_path
+                )
+
+        print_message(
+            f"Loaded all available handlers with a frequency of {freq}.",
+            "debug",
+        )
+        chosen_handlers = [handler for handler in available_handlers.values()]
+
+        return chosen_handlers
```

Comment:
This section handles when the user passes all handlers to the list of variables.

Reply:
It just loops through all of the default handlers and complex handlers that exist and adds them to the list of `chosen_handlers`.

This essentially mirrors a user passing all possible and valid default and complex handlers in `var_list`.

@chengzhuzhang and @TonyB9000, I don't think there is an issue with keeping this logic in considering the CWL workflows have predefined lists of variables. There may be `e3sm_to_cmip` users who use this feature.

If we do want to remove it due to feature bloat, we can release a new major reason since it is a breaking API change (`v2.0.0`).

Reply:
Understood.  That's the same reason I'm keeping the "all" variable handling in the DSM jobs/Generate*CMIP.py codes, even though the "if" will always be "false" the way I call it from my external calling scripts.